doc/source/Tutorials/Scripting/CreatingConfigurationFiles3/index.md

```diff
@@ -0,0 +1,154 @@
+```
+
+In this particular example of a node menu function, we don't modify any of the node's plugs, but we could – such as automatically setting its _Mode_ plug to 1. Any such additional node customization would go into the function, here.
+
+
+### Adding the functions to the node menu ###
+
+To finish the config, we add both functions to the node menu. This part is as simple as grabbing calling `append()` on the node menu itself:
+
+```eval_rst
+.. code-block:: python
+    :lineno-start: 15
+
+    nodeMenu = GafferUI.NodeMenu.acquire( application )
+    nodeMenu.append(
+    	path = "/Custom/MacbethTexture",
+    	nodeCreator = __macbethTexture,
+    	postCreator = __macbethTexturePostCreator,
+    	searchText = "MacbethTexture"
+    )
+```
+
+Line 14 merely grabs the node menu object from the `application` variable. Recall that we did the same in the custom bookmarks startup config.
+
+The `append()` convenience method is quite useful, as it wraps several other behind-the-scenes utility functions and makes the node's creation undoable. We pass four arguments to it:
+
+1. `path`: The path to the node within the node menu's hierarchy. Each position in the hierarchy is separated by a forward slash. If a parent position doesn't exist yet, it will be created automatically.
+2. `nodeCreator`: The node creation function. We don't pass any arguments to it.
+3. `postCreator`: The node post-creation function. We don't pass any arguments to this one, either, but both the node and the menu itself are implicitly passed to it; recall the `node` and `menu` keyword arguments we mentioned earlier.
+4. `searchText`: The menu entry's keyword search text. This is typically the same as the node name. Capitalization is ignored.
+
+That's all! You can add more customized nodes to this config with similar ease, as long as each is managed by its own separate function and registered to the node menu with an `append()` call.
+
+
+## Testing the Node Menu Entry ##
+
+Let's try testing the custom node. If you haven't already, save the startup config, then launch a new instance of Gaffer. In the _Graph Editor_, the new entry should appear in the node menu under _Custom_ > _MacbethTexture_, and will create a MacbethTexture node when selected.
+
+![The MacbethTexture node in the Graph Editor](images/tutorialMacbethTextureNode.png "The MacbethTexture node in the Graph Editor")
+
+
+## Demo: Procedural Macbeth Chart ##
+
+![Demo of the Macbeth texture assigned to a plane, to make a Macbeth chart](images/demoMacbethChart.png "Demo of the Macbeth texture assigned to a plane, to make a Macbeth chart")
+
+Here is a demo of the MacbethTexture node in action. We procedurally generate a Macbeth chart in a 3D scene by connecting the node's output to the _color_ input plug of a surface shader, and then assigning the shader to a plane mesh.
+
+```eval_rst
+    :download:`Download demo <demos/demoMacbethChart.gfr>`
```

Comment:
@medubelko Would you mind if we omitted this until we can fix up the issue with Reference paths?